src/util/triplea/MapXMLCreator/PlayerAndAlliancesPanel.java

```diff
@@ -0,0 +1,282 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class PlayerAndAlliancesPanel extends DynamicRowsPanel {
+		
+	private TreeSet<String> m_alliances = new TreeSet<String>();	
+	
+	public PlayerAndAlliancesPanel(final JPanel stepActionPanel)
+	{
+		super(stepActionPanel);
+	}
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel)
+	{
+		if (s_me == null || !(s_me instanceof PlayerAndAlliancesPanel))

```

Comment:
The Optional<..> is a signal to readers of the code, future developers who are trying to modify stuff (yourself 3 weeks after you stop touching this code and lose context, it's amazing how code looks different after a few weeks). specifically it's a signal that the value can be null.

Then instead of initializing the value to null, you initializes it to Optional.empty().  Then before accessing the object, you first check Optional.isPresent(), and if so, then you get the object. You now no longer have to worry about NPEs crashing your system. Here is an example:

```
public void foo( Optional<Bar> parameter ) {
   if( extraFunctions.isPresent() {

     extraFunction.get().doStuff( ... );
   }
}
```
Essentially the bottom line is to get rid of NPEs, you simply do not have nulls anywhere in your code. Instead you use objects that represent empty data, or worst case use Optional.

Guava added an optional type and it was later adopted in Java8. Use the java8 version instead of the guava one. Here is some more background: http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Optional.html, otherwise there are some pretty good write ups out there about why/how to use Optional.